keystrokes/src/components/Character/CharacterList.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import '../../stylesheets/character.css';
+import styleGuide from '../../stylesheets/style';
+import { ref } from '../../config/constants';
+import { List, ListItem, Typography, Divider } from '@material-ui/core';
+import CharacterCreationModal from './CharacterCreationModal';
+
+function CharacterList(props) {
+	const [characters, setCharacters] = useState({});
+	const [currentCharacter, setCurrentCharacter] = useState()
+	const [open, setOpen] = useState(false);
+	const classes = styleGuide();
+
+	useEffect(() => {
+		console.log(props)
+		setCharacters(props.userInfo.characters || {});
+		setCurrentCharacter(props.userInfo.currentCharacter);
+	}, [props]);
+
+	const handleOpen = () => {
+		setOpen(true);
+	}
+
+	async function handleSubmit(fields) {
+		const {name, characterClass, spec, race} = fields
+		let charactersSnapShot = await ref.child("/Characters");
+		let doesCharacterExist = await charactersSnapShot.child(name).once('value');
+		if (doesCharacterExist.exists()) {
+			console.log("error")
+			//throw error since name is currently id - replace with real key eventually
+			return;
+		} else {
+			charactersSnapShot
+			.child(name).set({
+				"class": characterClass, 
+				"race": race, 
+				"name": name, 
+				"specs": {}
+			});
+			charactersSnapShot
+			.child(name)
+			.child("specs")
+			.child(spec)
+			.set({
+				"spec": spec, 
+				"selected": true,
+				"configured": false
+			});
+		}
+		let snapshot = await ref.child(props.userPath);
+		let userCharacterData = await snapshot.once('value');
+		if (userCharacterData.exists()) {
+			snapshot
+				.child('characters').child(name) //name should probably be a real key
+				.set({ "name": name });
+		} else {
+			snapshot.set({
+					"currentCharacter": name,
+					"characters": {}
+			});
+			snapshot
+				.child('characters').child(name)
+				.set({ "name": name });
+		}
+		setCharacters({...characters, name: {"name": name}}) //immutabilityHelper here
+	}
+
+	return (
+		<React.Fragment>
+			{
+				open ?
```

Comment:
an alternative to this is simply 
`{open && <CharacterCreationModal/>}`

not sure if the modal needs to be in a fragment either.